src/main/java/com/funccover/CoverageTransformer.java

```
@@ -1,121 +1,125 @@
-            if (ct.isPrimitive() || ct.isArray() || ct.isAnnotation() || ct.isEnum() || ct.isInterface()) {
-                ct.detach();
-                return null;
-            }
-
-            // flag is true if instrumented
-            boolean flag = false;
-            
-            // Iterates over all methods and instruments them
-            for(CtMethod method: ct.getDeclaredMethods()) {
-                if (method.isEmpty()) {
-                    continue;
-                }
-                instrumentMethod(method, loader);
-                flag = true;
-            }
-
-            if (flag) {
-                result = ct.toBytecode();
-            }
-
-            // detach removes newly created class from cp to avoid unnecesarry memory consumption
-            ct.detach();
-        } catch (Throwable e) {
-            e.printStackTrace();
+    byte[] result = null;
+    try {
+      // Creates a new class with the given bytecode
+      CtClass ct = classPool.makeClass(new ByteArrayInputStream(classfileBuffer));
+
+      // checks if the class is already loaded
+      if (ct.isFrozen()) {
+        ct.detach();
+        return null;
+      }
+
+      // filter for instrumentation
+      if (ct.isPrimitive()
+          || ct.isArray()
+          || ct.isAnnotation()
+          || ct.isEnum()
+          || ct.isInterface()) {
+        ct.detach();
+        return null;
+      }
+
+      // flag is true if instrumented
+      boolean flag = false;
+
+      // Iterates over all methods and instruments them
+      for (CtMethod method : ct.getDeclaredMethods()) {
+        if (method.isEmpty()) {
+          continue;
         }
-        return result;
+        instrumentMethod(method, loader);
+        flag = true;
+      }
+
+      if (flag) {
+        result = ct.toBytecode();
+      }
+
+      // detach removes newly created class from cp to avoid unnecesarry memory consumption
+      ct.detach();
+    } catch (Throwable e) {
```

* Why "Throwable" instead of a more specific exception type?
* Why catch the exception at all? Are there "noise" exceptions where the behaviour of the instrumented program is not impacted?

---

Well, I thought it would be better not exit instrumentation if it fails somewhere. I can change it. I don't think there are noise exceptions that just doesn't mean anything. .

---

I'm undecided about whether to abort to let the user know that the instrumentation didn't work, or to just continue. Let's keep it as it (i.e. with catching & ignoring exceptions) is for now.